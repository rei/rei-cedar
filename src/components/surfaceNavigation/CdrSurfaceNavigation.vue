<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import CdrSurface from '../surface/CdrSurface.vue';
import type { surface } from '../../types/interfaces';
import { getSurfaceProps } from '../../utils/surface';

/** Navigation variant of CdrSurface */
defineOptions({ name: 'CdrSurNavigation' });

const props = withDefaults(defineProps<surface>(), { tag: 'div' });

const style = useCssModule();

// Compute surface props
const surfaceProps = computed(() => getSurfaceProps(props));
</script>

<template>
  <CdrSurface
    v-bind="surfaceProps"
    :class="style['cdr-surface-navigation']"
    :tag="props.tag"
  >
    <slot />
  </CdrSurface>
</template>

<style lang="scss" module src="./styles/CdrSurfaceNavigation.module.scss" />