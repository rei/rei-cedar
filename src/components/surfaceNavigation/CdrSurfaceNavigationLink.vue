<script setup lang="ts">
import { useCssModule } from 'vue';
import type { surfaceLink } from '../../types/interfaces';

/** Navigation-specific link for use within CdrSurfaceNavigation */

defineOptions({ name: 'CdrSurfaceNavigationLink' });

const props = withDefaults(defineProps<surfaceLink>(), {
  href: '#',
  tag: 'a',
});

const emit = defineEmits<{
  (e: 'click', event: MouseEvent): void
  (e: 'focus', event: FocusEvent): void
  (e: 'blur', event: FocusEvent): void
}>();

const handleClick = (event: MouseEvent) => {
  emit('click', event);
};

const handleFocus = (event: FocusEvent) => {
  emit('focus', event);
};

const handleBlur = (event: FocusEvent) => {
  emit('blur', event);
};

const style = useCssModule();
</script>

<template>
  <component
    v-bind="props"
    :is="props.tag"
    @click="handleClick"
    @focus="handleFocus"
    @blur="handleBlur"
    :class="style['cdr-surface-navigation-link']"
  >
    <slot />
  </component>
</template>

<style lang="scss" module src="./styles/CdrSurfaceNavigation.module.scss" />