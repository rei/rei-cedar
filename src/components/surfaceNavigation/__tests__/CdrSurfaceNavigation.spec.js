import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrSurfaceNavigation from '../CdrSurfaceNavigation.vue';
import CdrSurfaceNavigationLink from '../CdrSurfaceNavigationLink.vue';

describe('CdrSurface', () => {
  describe('snapshot test', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrSurfaceNavigation);
    });
    it('renders correctly', () => {
      wrapper.setProps({ palette: 'default' });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('component unit tests', () => {
    let wrapper;
    let link;
    beforeEach(() => {
      wrapper = mount(CdrSurfaceNavigation);
      link = mount(CdrSurfaceNavigationLink);
    });

    describe('default component with no configuration', () => {
      it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
      it('sets the default tag prop correctly', () => {
        expect(wrapper.element.tagName).toBe('DIV');
        expect(link.element.tagName).toBe('A');
      });
      it('default class added', () => {
        expect(wrapper.classes()).toEqual([
          'cdr-surface-navigation',
          'cdr-surface-navigation-radius-sharp--rest',
          'cdr-surface-navigation-shadow-flat--rest',
        ]);
      });
    });

    it('has data-palette attribute set to "default" by default', () => {
      expect(wrapper.attributes('data-palette')).toBe('default');
    });
    
  });
});
