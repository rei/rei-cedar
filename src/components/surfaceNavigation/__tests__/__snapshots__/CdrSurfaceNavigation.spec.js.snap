// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrSurface > component unit tests > default component with no configuration > renders correctly 1`] = `
<div
  class="cdr-surface-navigation cdr-surface-navigation-radius-sharp--rest cdr-surface-navigation-shadow-flat--rest"
  data-palette="default"
>
  <!-- @slot Where all default content should be placed. -->
  
  
</div>
`;

exports[`CdrSurface > snapshot test > renders correctly 1`] = `
<div
  class="cdr-surface-navigation cdr-surface-navigation-radius-sharp--rest cdr-surface-navigation-shadow-flat--rest"
  data-palette="default"
>
  <!-- @slot Where all default content should be placed. -->
  
  
</div>
`;
